app/controllers/facilities_controller.rb

```diff
@@ -1,46 +1,51 @@
 class FacilitiesController < ApplicationController
-    def index
-        @facilities = Facility.all
-    end
+  before_action :authenticate_user!, only: [:edit, :update, :destroy, :base_params, :create_params, :update_params]
+  def index
+    @facilities = Facility.all
+  end
     
-    def new
-        @facility = Facility.new
-    end
+  def new
+    @facility = Facility.new
+  end
     
-    def create
-        facility = Facility.new
-        facility.name = params[:facility][:name]
-        facility.city = params[:facility][:city]
-        facility.state = params[:facility][:state]
-        facility.zip = params[:facility][:zip]
-        facility.specialty = params[:facility][:specialty]
-        facility.save
-        redirect_to "/"
-        
-    end
+  def create
+    @facility = Facility.create(create_params)
+    @facility.save
```

Comment:
You don't need to call both `.create` and `.save`. The `.create` method is a combination of `.new` and `.save`, so you should either delete this line or change `.create` in the line above to `.new`.

Reply:
Deleted @facility,save.